test/to-have-state.spec.js

```diff
@@ -0,0 +1,71 @@
+import expect from './unexpected-enzyme';
+import { mount } from 'enzyme';
+import React, { Component } from 'react';
+
+class Counter extends Component {
+  constructor(props) {
+    super(props);
+
+    this.increment = this.increment.bind(this);
+
+    this.state = {
+      count: this.props.count
+    };
+  }
+
+  increment() {
+    this.setState(prev => ({
+      count: prev.count + 1
+    }));
+  }
+
+  render() {
+    const { count } = this.state;
+
+    return (
+      <div className="count">
+        {count}
+        <button className="increment" onClick={this.increment}>
+          Increment
+        </button>
+      </div>
+    );
+  }
+}
+
+describe('to-have-state', () => {
+  it('passes when the actual matches the expected', () => {
+    const wrapper = mount(<Counter count={0} />);
+
+    expect(wrapper, 'to have state', ['count']);
+
+    wrapper.find('button').simulate('click');
+
+    expect(wrapper, 'to have state', ['count']);
+  });
+
+  it('fails when the actual does not match the expected', () => {
+    expect(
+      () =>
+        expect(mount(<Counter count={0} />), 'to have state', [
+          'caunt',
+          'number'
+        ]),
+      'with error matching snapshot'
+    );
+  });
+
+  describe('when negated', () => {
+    it('passes when the actual does not match the expected', () => {
+      expect(mount(<Counter count={0} />), 'not to have state', ['foobar']);
```

Comment:
You see, I read this as the full state is an array with the string foobar in it.

Reply:
This is actually not possible, since React won't allow `state` to be an array, but I have changed it to `<ReactWrapper> to have state satisfying <assertion>`